pkg/porter/credentials.go

```diff
@@ -210,6 +206,43 @@ func (p *Porter) ShowCredential(opts CredentialShowOptions) error {
 // GetCredentialSourceValueAndType takes a given credentials.Source struct and
 // returns the source value itself as well as source type, e.g., 'path', 'env', etc.,
 // both in their string forms
-func GetCredentialSourceValueAndType(cs credentials.Source) (string, string) {
+func GetCredentialSourceValueAndType(cs credentials.Source) (value string, key string) {
 	return cs.Value, cs.Key
 }
+
+// CredentialDeleteOptions represent options for Porter's credential delete command
+type CredentialDeleteOptions struct {
+	Name string
+}
+
+// DeleteCredential deletes the credential set corresponding to the provided
+// names.
+func (p *Porter) DeleteCredential(opts CredentialDeleteOptions) error {
+	err := p.Credentials.Delete(opts.Name)
+	if err == crud.ErrRecordDoesNotExist {
+		fmt.Fprintln(p.Err, "credential set does not exist")
```

Comment:
Here is an example from another part of Porter that shows what I was trying to say with my previous comment. So we just need the check for `p.Debug` to be consistent.

https://github.com/deislabs/porter/blob/9c49056088dd07b954a0fb6238c6a60f5dfd4940/pkg/pkgmgmt/client/delete.go#L35-L37